src/coord/src/coord.rs

```diff
@@ -2202,6 +2202,42 @@ impl Coordinator {
         })
     }
 
+    /// Return the set of ids in a timedomain and verify timeline correctness.
+    ///
+    /// When a user starts a transaction, we need to prevent compaction of anything
+    /// they might read from. We use a heuristic of "anything in the same database
+    /// schemas with the same timeline as whatever the first query is".
+    fn timedomain_for(
+        &self,
+        source_ids: &[GlobalId],
+        source_timeline: &Option<Timeline>,
+        conn_id: u32,
+    ) -> Result<Vec<GlobalId>, CoordError> {
+        let mut timedomain_ids = self.catalog.schema_adjacent_relations(&source_ids, conn_id);
+
+        // Filter out ids from different timelines. The timeline code only verifies
+        // that the SELECT doesn't cross timelines. The schema-adjacent code looks
+        // for other ids in the same database schema. We need to remove ids from other
+        // timelines in the same time domain.
```

Comment:
should this say "timelines in the same database schema"?

Reply:
Yes. Just removed the whole sentence.